src/redist/targets/BundledTemplates.targets

```diff
@@ -1,55 +1,125 @@
+    <Bundled22Templates Include="Microsoft.DotNet.Web.Spa.ProjectTemplates" PackageVersion="$(AspNetCorePackageVersionFor22Templates)" />
+    <Bundled22Templates Include="NUnit3.DotNetNew.Template" PackageVersion="$(NUnit3Templates22PackageVersion)" />
+  </ItemGroup>
+  <ItemGroup>
+    <Bundled21Templates Include="Microsoft.DotNet.Common.ItemTemplates" PackageVersion="$(MicrosoftDotNetCommonItemTemplates21PackageVersion)" />
+    <Bundled21Templates Include="Microsoft.DotNet.Common.ProjectTemplates.2.1" PackageVersion="$(MicrosoftDotNetCommonProjectTemplates21PackageVersion)" />
+    <Bundled21Templates Include="Microsoft.DotNet.Test.ProjectTemplates.2.1" PackageVersion="$(MicrosoftDotNetTestProjectTemplates21PackageVersion)" />
+    <Bundled21Templates Include="Microsoft.DotNet.Web.ItemTemplates" PackageVersion="$(AspNetCorePackageVersionFor21Templates)" />
+    <Bundled21Templates Include="Microsoft.DotNet.Web.ProjectTemplates.2.1" PackageVersion="$(AspNetCorePackageVersionFor21Templates)" />
+    <Bundled21Templates Include="Microsoft.DotNet.Web.Spa.ProjectTemplates" PackageVersion="$(AspNetCorePackageVersionFor21Templates)" />
+    <Bundled21Templates Include="NUnit3.DotNetNew.Template" PackageVersion="$(NUnit3Templates21PackageVersion)" />
+  </ItemGroup>
 
-    <BundledTemplate Include="NUnit3.DotNetNew.Template" Version="$(NUnit3TemplatesVersion)" />
+  <ItemGroup>
+    <Bundled30Templates Update="@(Bundled30Templates)">
+      <NupkgPathRelativeToPackageRoot>%(Identity)/%(PackageVersion)/%(Identity).%(PackageVersion).nupkg</NupkgPathRelativeToPackageRoot>
+      <RestoredNupkgPath>$(NuGetPackageRoot)$([MSBuild]::ValueOrDefault('%(NupkgPathRelativeToPackageRoot)', '').ToLower())</RestoredNupkgPath>
+      <Version>[%(PackageVersion)]</Version>
+    </Bundled30Templates>
+  </ItemGroup>
+  <ItemGroup>
+    <Bundled22Templates Update="@(Bundled22Templates)">
+      <NupkgPathRelativeToPackageRoot>%(Identity)/%(PackageVersion)/%(Identity).%(PackageVersion).nupkg</NupkgPathRelativeToPackageRoot>
+      <RestoredNupkgPath>$(NuGetPackageRoot)$([MSBuild]::ValueOrDefault('%(NupkgPathRelativeToPackageRoot)', '').ToLower())</RestoredNupkgPath>
+      <Version>[%(PackageVersion)]</Version>
+    </Bundled22Templates>
+  </ItemGroup>
+  <ItemGroup>
+    <Bundled21Templates Update="@(Bundled21Templates)">
+      <NupkgPathRelativeToPackageRoot>%(Identity)/%(PackageVersion)/%(Identity).%(PackageVersion).nupkg</NupkgPathRelativeToPackageRoot>
+      <RestoredNupkgPath>$(NuGetPackageRoot)$([MSBuild]::ValueOrDefault('%(NupkgPathRelativeToPackageRoot)', '').ToLower())</RestoredNupkgPath>
+      <Version>[%(PackageVersion)]</Version>
+    </Bundled21Templates>
   </ItemGroup>
 
-  <!-- Restore bundled templates via PackageReference -->
   <ItemGroup>
-    <PackageReference Include="@(BundledTemplate)" />
+    <PackageDownload Include="@(Bundled22Templates);@(Bundled21Templates)" />
```

Comment:
Why are we using PackageDownload for 2.1/2.2?

Reply:
Using 'PackageReference' fails to download the lower version of the same package ID; 'PackageDownload' does not have this limitation.

Reply:
I don't remember the exact design of PackageDownload, if NuGet intends to restrict this to certain kinds of packages or not. Worth confirming, just to be safe and make sure we will not be broken later. Also, worried about getting one set of packages one way and another one in a different way.

Reply:
FWIW, if we are just using a NuGet package as a mechanism to transport a set of files, then I think `PackageDownload` makes more sense.  We might be able to simplify pieces of our build process that currently build a separate "download package" project multiple times to download different packages with `PackageDownload`.

Reply:
Awesome. I wanted confirmation the contract with NuGet was clear. Seems like it is.